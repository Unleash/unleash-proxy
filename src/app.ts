import compression from 'compression';
import express, { Application } from 'express';
import cors from 'cors';
import Client, { IClient } from './client';
import { createProxyConfig, IProxyOption } from './config';

import UnleashProxy from './unleash-proxy';

const corsOptions = {
    exposedHeaders: 'ETag',
    maxAge: 172800,
};

export function createApp(
    options: IProxyOption,
    unleashClient?: IClient,
    app: Application = express(),
): Application {
    const config = createProxyConfig(options);
    const client = unleashClient || new Client(config);

    const proxy = new UnleashProxy(client, config);

    app.disable('x-powered-by');
    app.set('trust proxy', config.trustProxy);
    app.use(cors(corsOptions));

    app.use(compression());

    app.use(
        `${config.proxyBasePath}/proxy`,
        cors(corsOptions),
        express.json(),
        proxy.middleware,
    );
    return app;
}

module.exports = { createApp };
