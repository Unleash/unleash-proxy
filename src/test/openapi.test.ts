import request from 'supertest';
import { Application } from 'express';
import { createApp } from '../app';
import MockClient from './client.mock';

const unleashUrl = 'http://localhost:4242/test';
const unleashApiToken = 'unleashApiToken';
const proxySecrets = ['proxySecrets'];
let app: Application;

beforeEach(() => {
    app = createApp(
        { proxySecrets, unleashUrl, unleashApiToken, enableOAS: true },
        new MockClient([]),
    );
});

test('should serve the OpenAPI UI', async () =>
    request(app)
        .get('/docs/openapi/')
        .expect(200)
        .then((response) => expect(response.text).toMatchSnapshot()));

test('should serve the OpenAPI spec', async () =>
    request(app)
        .get('/docs/openapi.json')
        .expect('Content-Type', /json/)
        .expect(200)
        .expect((res) => {
            // The version field is not set when running jest without yarn/npm.
            delete res.body.info.version;
            // This test will fail whenever there's a change to the API spec.
            // If the change is intended, update the snapshot with `jest -u`.
            expect(res.body).toMatchSnapshot();
        }));
