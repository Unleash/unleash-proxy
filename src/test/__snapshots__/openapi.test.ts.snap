// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should serve the OpenAPI UI 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <title>Swagger UI</title>
    <meta charset=\\"utf-8\\"/>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <style>
      html {
          box-sizing: border-box;
          overflow: -moz-scrollbars-vertical;
          overflow-y: scroll;
      }
      *,
      *:before,
      *:after {
          box-sizing: inherit;
      }
      body {
        margin: 0;
        padding: 0;
        background: #fafafa;
      }
    </style>
    
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"./swagger-ui.css\\" >
    
  </head>
  <body>
    
      <div id=\\"swagger-ui\\"></div>
      <script src=\\"./swagger-ui-bundle.js\\"></script>
      <script src=\\"./swagger-ui-standalone-preset.js\\"></script>
      <script src=\\"./swagger-ui-init.js\\"></script>
    
  </body>
</html>
  "
`;

exports[`should serve the OpenAPI spec 1`] = `
Object {
  "components": Object {
    "schemas": Object {
      "apiRequestSchema": Object {
        "additionalProperties": false,
        "properties": Object {
          "features": Object {
            "items": Object {
              "additionalProperties": false,
              "properties": Object {
                "description": Object {
                  "type": "string",
                },
                "enabled": Object {
                  "type": "boolean",
                },
                "impressionData": Object {
                  "type": "boolean",
                },
                "name": Object {
                  "type": "string",
                },
                "stale": Object {
                  "type": "boolean",
                },
                "strategies": Object {
                  "items": Object {
                    "additionalProperties": false,
                    "properties": Object {
                      "constraints": Object {
                        "items": Object {
                          "additionalProperties": false,
                          "properties": Object {
                            "contextName": Object {
                              "type": "string",
                            },
                            "operator": Object {
                              "description": "One of IN,NOT_IN,STR_ENDS_WITH,STR_STARTS_WITH,STR_CONTAINS,NUM_EQ,NUM_GT,NUM_GTE,NUM_LT,NUM_LTE,DATE_AFTER,DATE_BEFORE,SEMVER_EQ,SEMVER_GT,SEMVER_LT",
                              "type": "string",
                            },
                            "values": Object {
                              "items": Object {
                                "type": "string",
                              },
                              "type": "array",
                            },
                          },
                          "required": Array [
                            "contextName",
                            "operator",
                          ],
                          "type": "object",
                        },
                        "type": "array",
                      },
                      "name": Object {
                        "type": "string",
                      },
                      "parameters": Object {
                        "type": "object",
                      },
                    },
                    "required": Array [
                      "name",
                      "constraints",
                      "parameters",
                    ],
                    "type": "object",
                  },
                  "type": "array",
                },
                "variants": Object {
                  "items": Object {
                    "additionalProperties": false,
                    "properties": Object {
                      "name": Object {
                        "type": "string",
                      },
                      "overrides": Object {
                        "items": Object {
                          "additionalProperties": false,
                          "properties": Object {
                            "contextName": Object {
                              "type": "string",
                            },
                            "values": Object {
                              "items": Object {
                                "type": "string",
                              },
                              "type": "array",
                            },
                          },
                          "required": Array [
                            "contextName",
                            "values",
                          ],
                          "type": "object",
                        },
                        "type": "array",
                      },
                      "payload": Object {
                        "additionalProperties": false,
                        "properties": Object {
                          "type": Object {
                            "type": "string",
                          },
                          "value": Object {
                            "type": "string",
                          },
                        },
                        "required": Array [
                          "type",
                          "value",
                        ],
                        "type": "object",
                      },
                      "stickiness": Object {
                        "type": "string",
                      },
                      "weight": Object {
                        "type": "number",
                      },
                    },
                    "required": Array [
                      "name",
                      "weight",
                      "overrides",
                    ],
                    "type": "object",
                  },
                  "type": "array",
                },
              },
              "required": Array [
                "name",
              ],
              "type": "object",
            },
            "type": "array",
          },
          "version": Object {
            "enum": Array [
              2,
            ],
            "type": "integer",
          },
        },
        "required": Array [
          "features",
          "version",
        ],
        "type": "object",
      },
      "featureSchema": Object {
        "additionalProperties": false,
        "properties": Object {
          "enabled": Object {
            "type": "boolean",
          },
          "impressionData": Object {
            "type": "boolean",
          },
          "name": Object {
            "type": "string",
          },
          "variant": Object {
            "additionalProperties": false,
            "properties": Object {
              "enabled": Object {
                "type": "boolean",
              },
              "name": Object {
                "type": "string",
              },
              "payload": Object {
                "additionalProperties": false,
                "properties": Object {
                  "type": Object {
                    "enum": Array [
                      "string",
                    ],
                    "type": "string",
                  },
                  "value": Object {
                    "type": "string",
                  },
                },
                "required": Array [
                  "type",
                  "value",
                ],
                "type": "object",
              },
            },
            "required": Array [
              "name",
              "enabled",
            ],
            "type": "object",
          },
        },
        "required": Array [
          "name",
          "enabled",
          "impressionData",
        ],
        "type": "object",
      },
      "featuresSchema": Object {
        "additionalProperties": false,
        "properties": Object {
          "toggles": Object {
            "items": Object {
              "additionalProperties": false,
              "properties": Object {
                "enabled": Object {
                  "type": "boolean",
                },
                "impressionData": Object {
                  "type": "boolean",
                },
                "name": Object {
                  "type": "string",
                },
                "variant": Object {
                  "additionalProperties": false,
                  "properties": Object {
                    "enabled": Object {
                      "type": "boolean",
                    },
                    "name": Object {
                      "type": "string",
                    },
                    "payload": Object {
                      "additionalProperties": false,
                      "properties": Object {
                        "type": Object {
                          "enum": Array [
                            "string",
                          ],
                          "type": "string",
                        },
                        "value": Object {
                          "type": "string",
                        },
                      },
                      "required": Array [
                        "type",
                        "value",
                      ],
                      "type": "object",
                    },
                  },
                  "required": Array [
                    "name",
                    "enabled",
                  ],
                  "type": "object",
                },
              },
              "required": Array [
                "name",
                "enabled",
                "impressionData",
              ],
              "type": "object",
            },
            "type": "array",
          },
        },
        "required": Array [
          "toggles",
        ],
        "type": "object",
      },
      "lookupTogglesSchema": Object {
        "properties": Object {
          "context": Object {
            "properties": Object {
              "appName": Object {
                "type": "string",
              },
              "environment": Object {
                "type": "string",
              },
              "properties": Object {
                "additionalProperties": Object {
                  "anyOf": Array [
                    Object {
                      "type": "string",
                    },
                    Object {
                      "type": "number",
                    },
                  ],
                },
                "example": Object {
                  "betaTester": "true",
                  "region": "Africa",
                },
                "type": "object",
              },
              "remoteAddress": Object {
                "type": "string",
              },
              "sessionId": Object {
                "type": "string",
              },
              "userId": Object {
                "type": "string",
              },
            },
            "type": "object",
          },
          "toggles": Object {
            "example": Array [
              "myToggle",
              "yourToggle",
            ],
            "items": Object {
              "type": "string",
            },
            "type": "array",
          },
        },
        "type": "object",
      },
      "registerClientSchema": Object {
        "properties": Object {
          "appName": Object {
            "description": "Name of the application using Unleash",
            "type": "string",
          },
          "environment": Object {
            "deprecated": true,
            "type": "string",
          },
          "instanceId": Object {
            "description": "Instance id for this application (typically hostname, podId or similar)",
            "type": "string",
          },
          "interval": Object {
            "description": "At which interval, in milliseconds, will this client be expected to send metrics",
            "type": "number",
          },
          "sdkVersion": Object {
            "description": "Optional field that describes the sdk version (name:version)",
            "type": "string",
          },
          "started": Object {
            "description": "When this client started. Should be reported as ISO8601 time.",
            "oneOf": Array [
              Object {
                "format": "date-time",
                "type": "string",
              },
              Object {
                "type": "number",
              },
            ],
          },
          "strategies": Object {
            "description": "List of strategies implemented by this application",
            "items": Object {
              "type": "string",
            },
            "type": "array",
          },
        },
        "required": Array [
          "appName",
          "interval",
          "started",
          "strategies",
        ],
        "type": "object",
      },
      "registerMetricsSchema": Object {
        "properties": Object {
          "appName": Object {
            "type": "string",
          },
          "bucket": Object {
            "properties": Object {
              "start": Object {
                "format": "date-time",
                "type": "string",
              },
              "stop": Object {
                "format": "date-time",
                "type": "string",
              },
              "toggles": Object {
                "additionalProperties": Object {
                  "properties": Object {
                    "no": Object {
                      "minimum": 0,
                      "type": "integer",
                    },
                    "variants": Object {
                      "additionalProperties": Object {
                        "minimum": 0,
                        "type": "integer",
                      },
                      "type": "object",
                    },
                    "yes": Object {
                      "minimum": 0,
                      "type": "integer",
                    },
                  },
                  "type": "object",
                },
                "example": Object {
                  "myCoolToggle": Object {
                    "no": 42,
                    "variants": Object {
                      "blue": 6,
                      "green": 15,
                      "red": 46,
                    },
                    "yes": 25,
                  },
                  "myOtherToggle": Object {
                    "no": 100,
                    "yes": 0,
                  },
                },
                "type": "object",
              },
            },
            "required": Array [
              "start",
              "stop",
              "toggles",
            ],
            "type": "object",
          },
          "environment": Object {
            "type": "string",
          },
          "instanceId": Object {
            "type": "string",
          },
        },
        "required": Array [
          "appName",
          "instanceId",
          "bucket",
        ],
        "type": "object",
      },
      "unleashContextSchema": Object {
        "properties": Object {
          "appName": Object {
            "type": "string",
          },
          "environment": Object {
            "type": "string",
          },
          "properties": Object {
            "additionalProperties": Object {
              "anyOf": Array [
                Object {
                  "type": "string",
                },
                Object {
                  "type": "number",
                },
              ],
            },
            "example": Object {
              "betaTester": "true",
              "region": "Africa",
            },
            "type": "object",
          },
          "remoteAddress": Object {
            "type": "string",
          },
          "sessionId": Object {
            "type": "string",
          },
          "userId": Object {
            "type": "string",
          },
        },
        "type": "object",
      },
      "variantSchema": Object {
        "additionalProperties": false,
        "properties": Object {
          "enabled": Object {
            "type": "boolean",
          },
          "name": Object {
            "type": "string",
          },
          "payload": Object {
            "additionalProperties": false,
            "properties": Object {
              "type": Object {
                "enum": Array [
                  "string",
                ],
                "type": "string",
              },
              "value": Object {
                "type": "string",
              },
            },
            "required": Array [
              "type",
              "value",
            ],
            "type": "object",
          },
        },
        "required": Array [
          "name",
          "enabled",
        ],
        "type": "object",
      },
    },
    "securitySchemes": Object {
      "apiKey": Object {
        "in": "header",
        "name": "authorization",
        "type": "apiKey",
      },
    },
  },
  "info": Object {
    "title": "Unleash Proxy API",
  },
  "openapi": "3.0.3",
  "paths": Object {
    "/proxy": Object {
      "get": Object {
        "description": "This endpoint returns the list of feature toggles that the proxy evaluates to enabled for the given context. Context values are provided as query parameters.",
        "parameters": Array [
          Object {
            "description": "Your application's name",
            "in": "query",
            "name": "appName",
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "description": "The current user's ID",
            "in": "query",
            "name": "userId",
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "description": "The current session's ID",
            "in": "query",
            "name": "sessionId",
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "description": "Your application's IP address",
            "in": "query",
            "name": "remoteAddress",
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "description": "Additional (custom) context fields",
            "example": Object {
              "betaTester": "true",
              "region": "Africa",
            },
            "explode": true,
            "in": "query",
            "name": "properties",
            "schema": Object {
              "additionalProperties": Object {
                "type": "string",
              },
              "type": "object",
            },
            "style": "form",
          },
        ],
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "$ref": "#/components/schemas/featuresSchema",
                },
              },
            },
            "description": "The list of enabled toggles for the provided context.",
          },
          "401": Object {
            "description": "Authorization information is missing or invalid.",
          },
          "500": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "example": Object {
                    "error": "Whoops! We dropped the ball on this one (an unexpected error occurred): Cannot read properties of undefined (reading 'includes')",
                  },
                  "properties": Object {
                    "error": Object {
                      "type": "string",
                    },
                  },
                  "required": Array [
                    "error",
                  ],
                  "type": "object",
                },
              },
            },
            "description": "Something went wrong on the server side and we were unable to recover. If you have custom strategies loaded, make sure they don't throw errors.",
          },
          "503": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "example": "The Unleash Proxy has not connected to the Unleash API and is not ready to accept requests yet.",
                  "type": "string",
                },
              },
            },
            "description": "The proxy isn't ready to accept requests yet.",
          },
        },
        "summary": "Retrieve enabled feature toggles for the provided context.",
        "tags": Array [
          "Proxy client",
        ],
      },
      "post": Object {
        "description": "This endpoint accepts a JSON object with \`context\` and \`toggles\` properties. The Proxy will use the provided context values and evaluate the toggles provided in the \`toggle\` property. It returns the toggles that evaluate to false. As such, the list it returns is always a subset of the toggles you provide it.",
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "$ref": "#/components/schemas/lookupTogglesSchema",
              },
            },
          },
        },
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "$ref": "#/components/schemas/featuresSchema",
                },
              },
            },
            "description": "The list of enabled toggles for the provided context.",
          },
          "400": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "example": Object {
                    "error": "Request validation failed",
                    "validation": Array [
                      Object {
                        "dataPath": ".body",
                        "keyword": "required",
                        "message": "should have required property 'appName'",
                        "params": Object {
                          "missingProperty": "appName",
                        },
                        "schemaPath": "#/components/schemas/registerMetricsSchema/required",
                      },
                    ],
                  },
                  "properties": Object {
                    "error": Object {
                      "type": "string",
                    },
                    "validation": Object {
                      "items": Object {
                        "type": "object",
                      },
                      "type": "array",
                    },
                  },
                  "required": Array [
                    "error",
                  ],
                  "type": "object",
                },
              },
            },
            "description": "The provided request data is invalid.",
          },
          "401": Object {
            "description": "Authorization information is missing or invalid.",
          },
          "500": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "example": Object {
                    "error": "Whoops! We dropped the ball on this one (an unexpected error occurred): Cannot read properties of undefined (reading 'includes')",
                  },
                  "properties": Object {
                    "error": Object {
                      "type": "string",
                    },
                  },
                  "required": Array [
                    "error",
                  ],
                  "type": "object",
                },
              },
            },
            "description": "Something went wrong on the server side and we were unable to recover. If you have custom strategies loaded, make sure they don't throw errors.",
          },
          "503": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "example": "The Unleash Proxy has not connected to the Unleash API and is not ready to accept requests yet.",
                  "type": "string",
                },
              },
            },
            "description": "The proxy isn't ready to accept requests yet.",
          },
        },
        "summary": "Which of the provided toggles are enabled given the provided context?",
        "tags": Array [
          "Proxy client",
        ],
      },
    },
    "/proxy/client/features": Object {
      "get": Object {
        "description": "Returns the toggle configuration from the proxy's internal Unleash SDK. Use this to bootstrap other proxies and server-side SDKs. Requires you to provide one of the proxy's configured \`serverSideTokens\` for authorization.",
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "$ref": "#/components/schemas/apiRequestSchema",
                },
              },
            },
            "description": "The proxy's current feature toggle configuration. A list of feature toggles that is parseable by other server-side clients.",
          },
          "401": Object {
            "description": "Authorization information is missing or invalid.",
          },
          "503": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "example": "The Unleash Proxy has not connected to the Unleash API and is not ready to accept requests yet.",
                  "type": "string",
                },
              },
            },
            "description": "The proxy isn't ready to accept requests yet.",
          },
        },
        "summary": "Retrieve the proxy's current toggle configuration (as consumed by the internal client).",
        "tags": Array [
          "Server-side client",
        ],
      },
    },
    "/proxy/client/metrics": Object {
      "post": Object {
        "description": "This endpoint lets you register usage metrics with Unleash. Accepts either one of the proxy's configured \`serverSideTokens\` or one of its \`clientKeys\` for authorization.",
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "$ref": "#/components/schemas/registerMetricsSchema",
              },
            },
          },
        },
        "responses": Object {
          "200": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "example": "ok",
                  "type": "string",
                },
              },
            },
            "description": "The request was successful.",
          },
          "400": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "example": Object {
                    "error": "Request validation failed",
                    "validation": Array [
                      Object {
                        "dataPath": ".body",
                        "keyword": "required",
                        "message": "should have required property 'appName'",
                        "params": Object {
                          "missingProperty": "appName",
                        },
                        "schemaPath": "#/components/schemas/registerMetricsSchema/required",
                      },
                    ],
                  },
                  "properties": Object {
                    "error": Object {
                      "type": "string",
                    },
                    "validation": Object {
                      "items": Object {
                        "type": "object",
                      },
                      "type": "array",
                    },
                  },
                  "required": Array [
                    "error",
                  ],
                  "type": "object",
                },
              },
            },
            "description": "The provided request data is invalid.",
          },
          "401": Object {
            "description": "Authorization information is missing or invalid.",
          },
        },
        "summary": "Send usage metrics to Unleash.",
        "tags": Array [
          "Operational",
          "Server-side client",
        ],
      },
    },
    "/proxy/client/register": Object {
      "post": Object {
        "description": "This endpoint lets you register application with Unleash. Accepts either one of the proxy's configured \`serverSideTokens\` or one of its \`clientKeys\` for authorization.",
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "$ref": "#/components/schemas/registerClientSchema",
              },
            },
          },
        },
        "responses": Object {
          "200": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "example": "ok",
                  "type": "string",
                },
              },
            },
            "description": "The request was successful.",
          },
          "400": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "example": Object {
                    "error": "Request validation failed",
                    "validation": Array [
                      Object {
                        "dataPath": ".body",
                        "keyword": "required",
                        "message": "should have required property 'appName'",
                        "params": Object {
                          "missingProperty": "appName",
                        },
                        "schemaPath": "#/components/schemas/registerMetricsSchema/required",
                      },
                    ],
                  },
                  "properties": Object {
                    "error": Object {
                      "type": "string",
                    },
                    "validation": Object {
                      "items": Object {
                        "type": "object",
                      },
                      "type": "array",
                    },
                  },
                  "required": Array [
                    "error",
                  ],
                  "type": "object",
                },
              },
            },
            "description": "The provided request data is invalid.",
          },
          "401": Object {
            "description": "Authorization information is missing or invalid.",
          },
        },
        "summary": "Register clients with Unleash.",
        "tags": Array [
          "Operational",
          "Server-side client",
        ],
      },
    },
    "/proxy/health": Object {
      "get": Object {
        "description": "Returns a 200 OK if the proxy is ready to receive requests. Otherwise returns a 503 NOT READY.",
        "responses": Object {
          "200": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "example": "ok",
                  "type": "string",
                },
              },
            },
            "description": "The request was successful.",
          },
          "503": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "example": "The Unleash Proxy has not connected to the Unleash API and is not ready to accept requests yet.",
                  "type": "string",
                },
              },
            },
            "description": "The proxy isn't ready to accept requests yet.",
          },
        },
        "security": Array [],
        "summary": "Check whether the proxy is ready to serve requests yet.",
        "tags": Array [
          "Operational",
        ],
      },
    },
    "/proxy/internal-backstage/prometheus": Object {
      "get": Object {
        "description": "Returns a 200 and valid Prometheus text syntax if the proxy is ready to receive requests. Otherwise returns a 503 NOT READY.",
        "responses": Object {
          "200": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "example": "metric_name 1",
                  "type": "string",
                },
              },
            },
            "description": "The request was successful. Response in plain text format conforming to Prometheus exposition format",
          },
          "503": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "example": "The Unleash Proxy has not connected to the Unleash API and is not ready to accept requests yet.",
                  "type": "string",
                },
              },
            },
            "description": "The proxy isn't ready to accept requests yet.",
          },
        },
        "security": Array [],
        "summary": "Check whether the proxy is up and running",
        "tags": Array [
          "Operational",
        ],
      },
    },
  },
  "security": Array [
    Object {
      "apiKey": Array [],
    },
  ],
  "servers": Array [],
  "tags": Array [
    Object {
      "description": "Feature toggle endpoints intended to be consumed by proxy clients.",
      "name": "Proxy client",
    },
    Object {
      "description": "Feature toggle endpoints related to and intended to be consumed by server-side clients and other proxies.",
      "name": "Server-side client",
    },
    Object {
      "description": "Endpoints related to operating the Unleash proxy.",
      "name": "Operational",
    },
  ],
}
`;
